autorest/autorest.go

```diff
@@ -130,3 +131,20 @@ func NewPollingRequest(resp *http.Response, cancel <-chan struct{}) (*http.Reque
 
 	return req, nil
 }
+
+// NewPollingRequestWithContext allocates and returns a new http.Request with the specified context to poll for the passed response.
+func NewPollingRequestWithContext(ctx context.Context, resp *http.Response) (*http.Request, error) {
+	location := GetLocation(resp)
+	if location == "" {
+		return nil, NewErrorWithResponse("autorest", "NewPollingRequest", resp, "Location header missing from response that requires polling")
```

Comment:
>NewPollingRequest [](start = 48, length = 17)

WithContext?

Reply:
It's basically a copy of NewPollingRequest() that takes a context param (see the preceding func def).

Reply:
I mean `NewErrorWithResponse` second parameter should have a different value

Reply:
Derp, will fix.